src/lib/core/style/_elevation.scss

```diff
@@ -151,6 +151,14 @@ $_mat-elevation-prefix: 'mat-elevation-z';
               #{map-get(_get-ambient-map($color, $opacity), $zValue)};
 }
 
+@mixin mat-theme-elevation($zValue, $theme, $opacity: $mat-elevation-opacity) {
+  $foreground: map-get($theme, foreground);
+  $elevation-color: map-get($foreground, elevation);
```

Comment:
Typically the background and foreground palettes are fixed regardless of theme; how would you go about changing the color of the shadow?

Reply:
I took the position that wanting to change the elevation color was a fairly advanced use case, so it would be admissible for it to be changed by providing `angular-material-theme` with a fully custom theme map (as opposed to using the `mat-light-theme` and `mat-dark-theme` functions). This is what I do in a project where I want full control over the colours used in the app, however it has just occurred to me that it may not be part of the public API(?).

An alternative solution could be to provide it as an optional fourth argument to the `mat-light-theme` and `mat-dark-theme` functions (with default value `black`), and then to convert the `$mat-light-theme-foreground` and `$mat-dark-theme-foreground` variables into functions that take the elevation color as their only argument.

Reply:
@josephperrott and I took a closer look at this, and found that we're going to have some cases in Google where it will be more common to want to customize the shadow color based on the theme colors, so we probably want to make it more ergonomic to customize the shadow based on that.

We'd also want to avoid adding a lot of excess css to the theme so most people wouldn't end up using. I'm still thinking about a good way to approach it.

Reply:
@jelbourn Perhaps the additional parameter to the `theme` functions could be used instead?

An alternative way to avoid adding CSS to the themes could be to remove all internal usages of the mixins, and only to rely on the `.mat-elevation-z?` classes for applying elevation to Material components. This approach would actually reduce the amount of CSS overall, as only the output of

```
 @for $zValue from 0 through 24 {
    .#{$_mat-elevation-prefix}#{$zValue} {
      @include mat-theme-elevation($zValue, $theme);
    }
  }
```

would be rendered (component stylesheets would be smaller as they no longer include the box shadow rules).

Reply:
I like the idea of using the common set of css classes for everything. We would have to make sure they're still easily overridable, though. 

I guess what this PR is really about is being able to customize the _base_ shadow color, which makes sense to be in the foreground palette. If we want to set a shadow based on another palette, I don't see a way around explicitly passing the color in the component's theme file.